rfcs/web_extensions.md

```diff
@@ -0,0 +1,119 @@
+# RFC #219: Add support for WebExtensions in WPT
+
+## Summary
+
+The WebExtensions API extends the capabilities of the browser. Adding support
+for WebExtensions in the web-platform-tests will increase interoperability
+and will help drive the standardization of this API.
+
+This RFC proposes adding a new `testharness.js` wrapper type, `.extension.js`, to handle
+testing this API, in addition to using `testdriver.js` to load and unload extensions.
+   
+##  Details
+  
+Using `testdriver.js`, we added support for testing the WebExtensions API by loading
+a web extension designed to test the functionality of a specific API.
+
+These tests won't leverage `testharness.js` directly since most of the test execution
+is handled within the extension, via the `browser.test` API. We’ve elected to use
+this API since all participating browser vendors use `browser.test` internally and
+can easily port over existing tests to the web-platform-tests.  
+
+The `browser.test` API will also be used to send messages back to the web page to notify the testharness
+when a test has started and finished. For these cases, it is up to the user agent to send the message
+"test-started" and "test-finished", respectively. In addition, when recording the result for 
+a `browser.test.assert...()` method, it is up to the user agent to send the message "assert"
+or "assert-equality". 
+
+Because these tests don’t leverage `testharness.js` directly, we’ve introduced a new
+wrapper type, `.extension.js`, that will create the necessary boilerplate to
+convert these messages into the corresponding assertions in the `testharness.js`.
+   
+A proposed patch is available at https://github.com/web-platform-tests/wpt/pull/50648.
+
+##  `browser.test` API
```

Comment:
how does the browser know if the `browser.test` globals need to be exposed?

Reply:
In Safari, we expose the `browser.test` namespace when in a testing environment. This is true for WPT tests. Chrome and Firefox have agreed to do the same.

Reply:
Does it mean that the extension-specific globals would be exposed for non-extension WPT tests? Is the `browser` global variable extension-specific or is it used for other web features too? I wonder if it could cause problems for other WPT tests by exposing an unexpected global? given there the proposal now includes a feature flag for extensions perhaps browsers should only expose the globals if the feature=extensions is on? cc @oliverdunk if you have some more details on how Chromium would expose the test interface, it would be useful.